### PR TITLE
chore: upgrade libs, node 20

### PR DESCRIPTION
**Pull request by sammcj**
Upgraded repo to support current stable nodeJS and libraries.

- node 16 -> node 20
- actions-github v5 -> v6
- corrected input types (true != "true")

### CODE REVIEW

**Comment by iamludal** [comment]
Hi @aurelien-baudet , any updates on this PR please? Thanks.

File: action.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ inputs:
   display-workflow-run-url:
     description: 'Get the URL of the triggered workflow and display it in logs (useful to follow the progress of the triggered workflow)'
     required: false
-    default: true
```
Comment (aurelien-baudet) [inline action.yaml:26]: Why did you add quotes ? It should be a boolean value

File: action.yaml [hunk 2]
```diff
@@ -35,22 +35,22 @@ inputs:
   wait-for-completion:
     description: 'Block until the triggered workflow has finished'
     required: false
-    default: true
```
Comment (aurelien-baudet) [inline action.yaml:38]: Same here, why ?

**Review by aurelien-baudet — COMMENTED** [review]
It seems ok, just a question

File: action.yaml (hunk 1 above)
Comment (figroc) [reply action.yaml:26]: Should this line be changed? https://github.com/aurelien-baudet/workflow-dispatch/blob/00dba38060567621bfc0f88e7fc19a4932f4312f/src/utils.ts#L44

File: action.yaml (hunk 2 above)
Comment (figroc) [reply action.yaml:38]: And this: https://github.com/aurelien-baudet/workflow-dispatch/blob/00dba38060567621bfc0f88e7fc19a4932f4312f/src/utils.ts#L49

**Comment by VincentLanglet** [comment]
What's the status of the PR ?

Also, there is a warning with the action:
```
Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

**Comment by MugishaU** [comment]
Also wondering if there is any update on this?

File: action.yaml (hunk 1 above)
Comment (aurelien-baudet) [reply action.yaml:26]: Not really, why ?

Comment (Djontleman) [reply action.yaml:26]: > It should be a boolean value

Looking at the [schema](https://json.schemastore.org/github-action.json), the type of `default` is `string`. However, I don't think it will actually change any behaviour as GHA likely coerces it into a string anyway. This [StackOverflow answer](https://stackoverflow.com/questions/19109912/yaml-do-i-need-quotes-for-strings-in-yaml) may also help explain.

It may be worth adding the quotes if only to avoid your editor showing a syntax error, but it should probably then use single-quotes to be consistent with the rest of the file.

> Should this line be changed?

I also don't think this line should be changed. Whether the input is processed as a string or boolean is dependent on whether `core.getInput` or `core.getBooleanInput` is used in the script: https://www.npmjs.com/package/@actions/core#inputsoutputs

**Comment by aurelien-baudet** [comment]
Sorry, I have no more time and energy to maintain this repository. I even don't have computer anymore. Someone of you should for this repository and merge the different pull requests.
Tell me who creates the fork and I will archive this repository but indicate which repository follows up

**Comment by iamludal** [comment]
Hi @aurelien-baudet, I have created a fork of your repository that I can maintain. https://github.com/iamludal/workflow-dispatch

**Comment by VincentLanglet** [comment]
> Hi @aurelien-baudet, I have created a fork of your repository that I can maintain. [iamludal/workflow-dispatch](https://github.com/iamludal/workflow-dispatch)

Does this project really need _another fork_ when there is already the original repository ?
https://github.com/benc-uk/workflow-dispatch

The original repository seems still maintained. Is there a reason to not for moving to it ?

**Comment by iamludal** [comment]
> > Hi @aurelien-baudet, I have created a fork of your repository that I can maintain. [iamludal/workflow-dispatch](https://github.com/iamludal/workflow-dispatch)
> 
> Does this project really need _another fork_ when there is already the original repository ? https://github.com/benc-uk/workflow-dispatch
> 
> The original repository seems still maintained. Is there a reason to not for moving to it ?

Yes, this for from @aurelien-baudet adds some inputs to wait for the workflow completion before exiting. In the original repository, the workflow is triggered and exits successfully immediately.

**Comment by VincentLanglet** [comment]
> Yes, this fork from @aurelien-baudet adds some inputs to wait for the workflow completion before exiting. In the original repository, the workflow is triggered and exits successfully immediately.

Thanks for the answer.

IMHO the best, if possible, would be to add those inputs in the original repository which are well maintained.

**Comment by aurelien-baudet** [comment]
Repository transferred to https://github.com/the-actions-org/workflow-dispatch

**Comment by fbiesse** [comment]
This issues were resolved in https://github.com/the-actions-org/workflow-dispatch/pull/18

**Comment by fbiesse** [comment]
These issues were resolved in https://github.com/the-actions-org/workflow-dispatch/pull/18
